src/JointPositionGoalController.cpp

```diff
@@ -0,0 +1,349 @@
+
+    void JointPositionGoalController::update(const ros::Time &time, const ros::Duration &period) {
+        handler_->update();
+        lcm_->handleTimeout(0);
+
+        double dt = (time - last_update_).toSec();
+        int64_t utime = (int64_t) (time.toSec() * 1000000.);
+
+        size_t number_of_joint_interfaces = positionJointHandles_.size();
+
+        // VAL_CORE_ROBOT_STATE
+        // push out the joint states for all joints we see advertised
+        // and also the commanded torques, for reference
+        bot_core::joint_state_t lcm_pose_msg;
+        lcm_pose_msg.utime = utime;
+        lcm_pose_msg.num_joints = (int16_t) number_of_joint_interfaces;
+        lcm_pose_msg.joint_name.assign(number_of_joint_interfaces, "");
+        lcm_pose_msg.joint_position.assign(number_of_joint_interfaces, (const float &) 0.);
+        lcm_pose_msg.joint_velocity.assign(number_of_joint_interfaces, (const float &) 0.);
+        lcm_pose_msg.joint_effort.assign(number_of_joint_interfaces, (const float &) 0.);
+
+        // VAL_COMMAND_FEEDBACK
+        bot_core::joint_state_t lcm_commanded_msg;
+        lcm_commanded_msg.utime = utime;
+        lcm_commanded_msg.num_joints = (int16_t) number_of_joint_interfaces;
+        lcm_commanded_msg.joint_name.assign(number_of_joint_interfaces, "");
+        lcm_commanded_msg.joint_position.assign(number_of_joint_interfaces, (const float &) 0.);
+        lcm_commanded_msg.joint_velocity.assign(number_of_joint_interfaces, (const float &) 0.);
+        lcm_commanded_msg.joint_effort.assign(number_of_joint_interfaces, (const float &) 0.);
+
+        // EST_ROBOT_STATE
+        // need to decide what message we're really using for state. for now,
+        // assembling this to make director happy
+        bot_core::robot_state_t lcm_state_msg;
+        lcm_state_msg.utime = utime;
+        lcm_state_msg.num_joints = (int16_t) number_of_joint_interfaces;
+        lcm_state_msg.joint_name.assign(number_of_joint_interfaces, "");
+        lcm_state_msg.joint_position.assign(number_of_joint_interfaces, (const float &) 0.);
+        lcm_state_msg.joint_velocity.assign(number_of_joint_interfaces, (const float &) 0.);
+        lcm_state_msg.joint_effort.assign(number_of_joint_interfaces, (const float &) 0.);
+        lcm_state_msg.pose.translation.x = 0.0;
+        lcm_state_msg.pose.translation.y = 0.0;
+        lcm_state_msg.pose.translation.z = 0.0;
+        lcm_state_msg.pose.rotation.w = 1.0;
+        lcm_state_msg.pose.rotation.x = 0.0;
+        lcm_state_msg.pose.rotation.y = 0.0;
+        lcm_state_msg.pose.rotation.z = 0.0;
+        lcm_state_msg.twist.linear_velocity.x = 0.0;
+        lcm_state_msg.twist.linear_velocity.y = 0.0;
+        lcm_state_msg.twist.linear_velocity.z = 0.0;
+        lcm_state_msg.twist.angular_velocity.x = 0.0;
+        lcm_state_msg.twist.angular_velocity.y = 0.0;
+        lcm_state_msg.twist.angular_velocity.z = 0.0;
+
+        // Iterate over all position-controlled joints
```

Comment:
Same here. Make a function called publishRobotStateToLCM

See how the Atlas driver is split up into different files